[DOCS] Clarify AI Assistant will spawn ML nodes (#3687)
elastic/observability-docs: docs/en/observability/observability-ai-assistant.asciidoc

** Azure OpenAI Service `gpt-4`(0613) or `gpt-4-32k`(0613) with API version `2023-07-01-preview` or more recent.
* The knowledge base requires a 4 GB {ml} node.

[IMPORTANT]

> ====
> In {ecloud} or {ece}, if you have Machine Learning autoscaling enabled, Machine Learning nodes will be started when using the knowledge base and AI Assistant. Therefore using these features will incur additional costs.
> ====
> 

Reviewer: ```suggestion
====
In {ecloud} or {ece}, if you have Machine Learning autoscaling enabled, Machine Learning nodes will be started when using the knowledge base and AI Assistant. Therefore using these features will incur additional costs.

Note that disabling the AI Assistant will not result in ML nodes being scaled down.
====

```

Author: I think that's great @dedemorton 
I would love to get confirmation by engineering whether ML nodes will scale down.
ATM, it seems that once ML jobs are stopped or finish, ML nodes can be automatically scaled down.

But AI Assistant is not using ML Jobs but other ML features...

Reviewer 2: Following up here. Did we ever figure out if ML nodes scale down? This comment https://github.com/elastic/support-tech-lead/issues/1392#issuecomment-2030864999 says we do.

Reviewer 3: Few things:

- I'm not sure what "disabling the AI Assistant" means. There are different ways to do this.
- When the user sets up the knowledge base (this is an explicit user action), we download and deploy ELSER (the best model for the current architecture).
- There is no way to disable or uninstall the knowledge base. If the user disables the AI Assistant (either the plugin, or the UI), the ELSER deployment stays up. We cannot "undeploy" ELSER - other parts of the Platform might be using it (and might originally have deployed it).

Reviewer: @dgieselaar Thanks for the info. Sorry I missed your comment when I was OOTO a few weeks ago. Do you want me to rewrite Luca's note using all the details you provided, or do you just want me to modify the note to indicate that:

- setting up the knowledge base may trigger an autoscaling event and incur additional costs
- there is no way to disable the knowledge base after it is installed [IMO saying this might worry users]

If you're not asking for a complete rewrite of the note, GitHub's suggest feature is a great way to indicate specific changes. Thanks!

** Azure OpenAI Service `gpt-4`(0613) or `gpt-4-32k`(0613) with API version `2023-07-01-preview` or more recent.
* The knowledge base requires a 4 GB {ml} node.

[IMPORTANT]
====

> In Elastic Cloud or ECE, if you have Machine Learning autoscaling enabled, Machine Learning nodes will be started when using the Knowledge Base and AI Assistant features.
> Therefore, it will incur into additional costs.

Reviewer: ```suggestion
In {ecloud} or {ece}, if you have Machine Learning autoscaling enabled, Machine Learning nodes will be started when using the knowledge base and AI Assistant. Therefore using these features will incur additional costs.
```
